components/Dialogs/examples/DialogsTallTextAlertExampleViewController.m

```diff
@@ -0,0 +1,99 @@
+
+@property(nonatomic, strong) MDCDialogTransitionController *transitionController;
+
+@end
+@implementation DialogsTallTextAlertExampleViewController
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+
+  // We must create and store a strong reference to the transitionController.
+  // A presented view controller will set this object as its transitioning delegate.
+  self.transitionController = [[MDCDialogTransitionController alloc] init];
+
+  MDCButton *presentButton = [[MDCButton alloc] initWithFrame:CGRectZero];
+  [presentButton setTitle:[NSString stringWithFormat:@"Dialog with large Urdu text"]
+                 forState:UIControlStateNormal];
+  [presentButton addTarget:self
+                    action:@selector(didTapPresent:)
+          forControlEvents:UIControlEventTouchUpInside];
+  [self.view addSubview:presentButton];
+  presentButton.translatesAutoresizingMaskIntoConstraints = NO;
+
+  [self.view addConstraints:@[
+    [NSLayoutConstraint constraintWithItem:self.view
+                                 attribute:NSLayoutAttributeCenterX
+                                 relatedBy:NSLayoutRelationEqual
+                                    toItem:presentButton
+                                 attribute:NSLayoutAttributeCenterX
+                                multiplier:1
+                                  constant:0],
+    [NSLayoutConstraint constraintWithItem:self.view
+                                 attribute:NSLayoutAttributeCenterY
+                                 relatedBy:NSLayoutRelationEqual
+                                    toItem:presentButton
+                                 attribute:NSLayoutAttributeCenterY
+                                multiplier:1
+                                  constant:0],
+  ]];
+}
+
+- (IBAction)didTapPresent:(id)sender {
+  NSString *networkFailureInUrdu = @"براہ کرم اپنا نیٹ ورک کنکشن چیک کریں اور دوبارہ کوشش کریں۔";
+  MDCAlertController *alert = [MDCAlertController alertControllerWithTitle:nil
+                                                                   message:networkFailureInUrdu];
+  NSString *urduFontName = @"NotoNastaliqUrdu";
+  UIFont *dialogBodyFont;
+  UIFont *dialogButtonFont;
+  if (@available(iOS 11, *)) {
+    // Noto Nastaliq Urdu was added in iOS 11, and is an extremely tall
+    // font for any given nominal point size.
+    dialogBodyFont = [UIFont fontWithName:urduFontName size:20.0];
+    dialogButtonFont = [UIFont fontWithName:urduFontName size:26.0];
+  } else {
+    dialogBodyFont = [UIFont systemFontOfSize:20.0];
+    dialogButtonFont = [UIFont systemFontOfSize:26.0];
+  }
+  alert.messageFont = dialogBodyFont;
+  alert.buttonFont = dialogButtonFont;
+
+  MDCAlertAction *retryAction = [MDCAlertAction actionWithTitle:@"دوبارہ کوشش کریں" handler:nil];
+  MDCAlertAction *cancelAction = [MDCAlertAction actionWithTitle:@"منسوخ کریں" handler:nil];
+  [alert addAction:retryAction];
+  [alert addAction:cancelAction];
```

Comment:
Can we theme the alert before presenting it?
I believe the recommended logic (that we should promote here) is to call our themers, and then override required values, like which fonts to use for buttons, after our themers are called.

Reply:
Does it mean `buttonFont` and `messageFont` APIs are deprecated or discouraged to be used?

Reply:
Opened https://github.com/material-components/material-components-ios/issues/7148 to track.